### ISSUE TITLE
ARM build?

### ISSUE THREAD

**Issue by ASpieler** [Lolliedieb/lolMiner-releases#294]
Feature Request: aarch64/armv8 build.
any chance of it happening?

**Comment by jgonzis** [comment]
> Feature Request: aarch64/armv8 build.
> any chance of it happening?

For which algo? Actually all algo are GPU orientated...

**Comment by Heasterian** [comment]
It's possible to use OpenCL acceleration in Termux so algos based on it could run even on phones with correct libraries.

**Comment by ASpieler** [comment]
I was actually referring to a build that would run _on_ aarch64/armv8 for GPU mining.
I have a micro computer running a Cortex-A72 processor SOC with a PCIe 1x lane.

**Comment by Lolliedieb** [comment]
The C++ part of the miner certainly could build for other architectures - not sure about the GPU part though. Can you drop more info about the system you want to use so I can make an estimate if this makes sense?

**Comment by ASpieler** [comment]
To clarify, the SOC is a Broadcom BCM2711, which which does have an internal VideoCore VI graphics processor, but the GPU i was hoping to mine with is a standard desktop graphics card (NVIDIA GeForce 1060/1660/2070/3060ti/etc.) attached to the SOC's PCIe 1x lane, so theoretically, the actual GPU code should not change much.

**Comment by yxvtri** [comment]
Please, Yes I do too because I have a TEGRA X1 with Linux aarch64 and armv8. :)

**Comment by ASpieler** [comment]
yes, that was the idea, but so far none of the many miner authors I've reached out to showed any interest in making a compatible build of their miners, and most of them seem to contain code that thinks qemu-static (used for testing/proof of concept) is trying to decompile them/subvert security, so I can't even do a proof of concept with most, to show that it would work.

P.S. was using qemu-static on the CM4 to see if the miners could be ran on it that way, with direct peripheral hardware access enabled.

**Comment by Saikatsaha1996** [comment]
> > Feature Request: aarch64/armv8 build.
> > any chance of it happening?
> 
> For which algo? Actually all algo are GPU orientated...

Arm opencl GPU miner posible... With android device also only for Mali GPU bez arm OpenCL Mali GPU can mine ... Adreno GPU can't... I already tried..

![Screenshot_2022-01-15-09-01-56-036_com termux](https://user-images.githubusercontent.com/72664192/160301452-2df6f0c2-effe-4ae5-ae53-acae7c4d071d.jpg)
